### PR TITLE
Adjust init time of journal

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
After journal we should update some counts in smartfs_dev, for example dev->releasecount
To calculate this, we have to finish smart_scan. So modify to initialize journal
before smart_scan execute.
So we don't need extra Map structure for loser anymore.

### CODE REVIEW

File: os/fs/driver/mtd/smart.c [hunk 1]
```diff
@@ -5484,14 +5448,17 @@ static int smart_journal_recovery(FAR struct smart_struct_s *dev, journal_log_t
 
 	/* For Release & Erase, Nothing for now, Retry Checkout */
 	case SMART_JOURNAL_TYPE_RELEASE:
+		return smart_journal_checkout_process(dev, log, address);
 	case SMART_JOURNAL_TYPE_ERASE:
-		break;
+		ret = smart_journal_checkout_process(dev, log, address);
+		if (ret = OK) {
```
Comment (jeongchanKim) [inline os/fs/driver/mtd/smart.c:5454]: Please modify to `ret == OK`.

File: os/fs/driver/mtd/smart.c [hunk 2]
```diff
@@ -2232,24 +2210,10 @@ static int smart_scan(FAR struct smart_struct_s *dev)
...
-			goto err_out;
-		}
-	}
-#endif
 	ret = OK;
 
 err_out:
-#ifdef CONFIG_MTD_SMART_JOURNALING
-	if (dev->sSeqLogMap != NULL) {
-		kmm_free(dev->sSeqLogMap);
```
Comment (jeongchanKim) [inline os/fs/driver/mtd/smart.c:2250]: There is a memory allocation for `rootdirdev`. But there is no free for failure case. Is it okay?

Comment (Taejun-Kwon) [reply os/fs/driver/mtd/smart.c:2250]: yes it should be kept. it is block driver's private data, all the time it should be kept until smartfs unmounted.

File: os/fs/driver/mtd/smart.c (hunk 1 above)
Comment (Taejun-Kwon) [reply os/fs/driver/mtd/smart.c:5454]: fixed.

**Pull request merged**